app/src/org/commcare/android/framework/EntityDetailFragment.java

```diff
@@ -70,8 +71,8 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
         // Note that some of this setup could be moved into onAttach if it would help performance
         Detail childDetail = getChildDetail();
-        NodeEntityFactory factory = new NodeEntityFactory(childDetail, asw.getEvaluationContext());
         TreeReference childReference = getChildReference();
+        NodeEntityFactory factory = new NodeEntityFactory(childDetail, this.getFactoryContext());

```

Comment:
It worries me that the reference retrieved above (line 74) and below (line 116) are the same, but coupled. Seems worth passing in rather than duplicating to future proof against synchronicity issues between the two.